src/utils/PolygonConnector.h

```diff
@@ -106,51 +195,285 @@ class PolygonConnector
+            for(size_t to_index = 0; to_index < to_polygons[poly_index].size(); ++to_index)
+            {
+                const Point to_pos1 =  getPosition(to_polygons[poly_index][to_index]);
+                const coord_t to_width1 = getWidth(to_polygons[poly_index][to_index]);
+                const Point to_pos2 =  getPosition(to_polygons[poly_index][(to_index + 1) % to_polygons[poly_index].size()]);
+                const coord_t to_width2 = getWidth(to_polygons[poly_index][(to_index + 1) % to_polygons[poly_index].size()]);
+                const coord_t smallest_to_width = std::min(to_width1, to_width2);
+
+                for(size_t from_index = 0; from_index < from_poly.size(); ++from_index)
+                {
+                    const Point from_pos1 = getPosition(from_poly[from_index]);
+                    const coord_t from_width1 = getWidth(from_poly[from_index]);
+                    const Point from_pos2 = getPosition(from_poly[(from_index + 1) % from_poly.size()]);
+                    const coord_t from_width2 = getWidth(from_poly[(from_index + 1) % from_poly.size()]);
+                    const coord_t smallest_from_width = std::min(from_width1, from_width2);
+
+                    //Try a naive distance first. Faster to compute, but it may estimate the distance too small.
+                    coord_t naive_dist = LinearAlg2D::getDistFromLine(from_pos1, to_pos1, to_pos2);
+                    if(naive_dist - from_width1 - smallest_to_width < line_width * max_gap)
+                    {
+                        const Point closest_point = LinearAlg2D::getClosestOnLineSegment(from_pos1, to_pos1, to_pos2);
+                        const coord_t width_at_closest = interpolateWidth(closest_point, to_polygons[poly_index][to_index], to_polygons[poly_index][(to_index + 1) % to_polygons[poly_index].size()]);
+                        const coord_t distance = vSize(closest_point - from_pos1) - from_width1 - width_at_closest; //Actual, accurate distance to the other polygon.
+                        if(distance < best_distance)
+                        {
+                            PolygonConnection<Polygonal> first_connection = PolygonConnection<Polygonal>(&from_poly, from_index, from_pos1, &to_polygons[poly_index], to_index, closest_point);
+                            std::optional<PolygonConnection<Polygonal>> second_connection = getSecondConnection(first_connection, (width_at_closest + from_width1) / 2);
+                            if(second_connection) //Second connection is also valid.
+                            {
+                                best_distance = distance;
+                                best_connection = first_connection;
+                                best_second_connection = second_connection;
+                            }
+                        }
+                    }
+
+                    //Also try the other way around: From the line segment of the from_poly to a vertex in the to_polygons.
```

Comment:
Yeah, could be done indeed. But it'd have to pass the 3 vertices (`from_pos1`, `to_pos1` and `to_pos1`) and have 3 output variables. It would either return `distance`, `first_connection` and `second_connection` and then repeat the updating of the best connection twice, or take `best_distance`, `best_connection` and `best_second_connection` by reference and modify them in-place which is extra complexity. Since it's relatively complex to follow for relatively little gain in maintainability, I think we can leave it like this for now. Unless someone feels strongly about it.